cmd/sti/main.go

```diff
@@ -59,14 +67,24 @@ func newCmdBuild(req *api.Request) *cobra.Command {
 		Short: "Build a new image",
 		Long:  "Build a new Docker image named <tag> from a source repository and base image.",
 		Run: func(cmd *cobra.Command, args []string) {
-			if len(args) < 3 {
+			// Attempt to restore the build command from the configuration file
+			config.Restore(req, cmd)
+
+			// If user specifies the arguments, then we override the stored ones
+			if len(args) >= 3 {
+				req.Source = args[0]
+				req.BaseImage = args[1]
+				req.Tag = args[2]
+			}
+
+			if !validRequest(req) {
```

Comment:
I still prefer to have method for this as the validation is pretty long line and does not explicitly say what it is doing. I revamped the method to just return the bool instead of having if, hope that is OK ;-)